### PR TITLE
Azure OpenAI: 2024-02-15-preview support 

### PR DESCRIPTION
**Pull request by ninghu**
This change to the Azure OpenAI SDK introduces the Text to Speech (TTS) functionality, providing users with the ability to convert written text into spoken words in a natural and lifelike manner. Alongside this significant addition, it also brings several improvements to the existing features to enhance user experience for developers.

### Features Added

- Text to speech using OpenAI TTS models is now supported. See [OpenAI's API reference](https://platform.openai.com/docs/api-reference/audio/createSpeech) or the [Azure OpenAI quickstart](https://learn.microsoft.com/en-us/azure/ai-services/openai/text-to-speech-quickstart) for detailed overview and background information.
- The new method `GetAudioSpeech` exposes this capability on `OpenAIClient`.
- Text-to-speech converts text into lifelike spoken audio in a chosen voice, together with other optional configurations.
- This method works for both Azure OpenAI and non-Azure `api.openai.com` client configurations

### Breaking Changes

"On Your Data" changes:

- Introduced a new type `AzureChatExtensionDataSourceResponseCitation` for a more structured representation of citation data.
- Correspondingly, updated `AzureChatExtensionsMessageContext`:
  - Replaced `Messages` with `Citations` of type `AzureChatExtensionDataSourceResponseCitation`.
  - Added `Intent` as a string type.
- Renamed "AzureCognitiveSearch" to "AzureSearch":
  - `AzureCognitiveSearchChatExtensionConfiguration` is now `AzureSearchChatExtensionConfiguration`.
  - `AzureCognitiveSearchIndexFieldMappingOptions` is now `AzureSearchIndexFieldMappingOptions`.
- Check the project README for updated code snippets.

### Other Changes

- New properties in `ChatCompletionsOptions`:
  - `EnableLogProbabilities`: Allows retrieval of log probabilities (REST: `logprobs`)
  - `LogProbabilitiesPerToken`: The number of most likely tokens to return per token (REST: `top_logprobs`)
- Introduced a new property in `CompletionsOptions`:
  - `Suffix`: Defines the suffix that follows the completion of inserted text (REST: `suffix`)
- Image generation response now includes content filtering details (specific to Azure OpenAI endpoint):
  - `ImageGenerationData.ContentFilterResults`: Information about the content filtering results. (REST: `content_filter_results`)
  - `ImageGenerationData.PromptFilterResults`: Information about the content filtering category (REST: `prompt_filter_results`)

### CODE REVIEW

File: sdk/openai/Azure.AI.OpenAI/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Release History
 
-## 1.0.0-beta.14 (Unreleased)
+## 1.0.0-beta.15 (Unreleased)
```
Comment (trrwilson) [inline sdk/openai/Azure.AI.OpenAI/CHANGELOG.md:3]: We can omit the "vNext" entry -- the last version was actually a sequencing error on my part and I believe we generally want the released release notes to just start with the current (released) version.

File: sdk/openai/Azure.AI.OpenAI/README.md [hunk 2]
```diff
@@ -693,6 +693,22 @@ Console.WriteLine($"Translation ({translation.Duration.Value.TotalSeconds}s):");
 Console.WriteLine(translation.Text);
 ```
 
+### Generate Speech with Text-to-Speech (TTS) models
+
+```C# Snippet:SpeechGeneration
+SpeechGenerationOptions speechOptions = new()
+{
+    Input = "Hello World",
```
Comment (trrwilson) [inline sdk/openai/Azure.AI.OpenAI/README.md:701]: (Totally optional, channeling some of the discussion from parallel conversations but can be deferred until later with no issues): if we'd like to, I think it's a defensible argument that `Input` on the new options type could be more descriptively named `Text`.

File: sdk/openai/Azure.AI.OpenAI/tests/CompletionsTests.cs [hunk 3]
```diff
@@ -108,7 +108,6 @@ public async Task AdvancedCompletionsOptions(Service serviceTarget)
                 GenerationSampleCount = 3,
                 Temperature = 0.75f,
                 User = "AzureSDKOpenAITests",
-                Echo = true,
```
Comment (trrwilson) [inline sdk/openai/Azure.AI.OpenAI/tests/CompletionsTests.cs:111]: was this removal due to the `instruct` model switch no longer supporting the property? I think that's fine if so; the schema still supports `echo` and it'd be nice if we still validated it, but given this is "legacy" status I think that can be deferred

File: sdk/openai/Azure.AI.OpenAI/tests/CompletionsTests.cs [hunk 4]
```diff
@@ -133,15 +132,48 @@ public async Task AdvancedCompletionsOptions(Service serviceTarget)
...
+            OpenAIClient client = GetTestClient(serviceTarget);
+            string deploymentOrModelName = GetDeploymentOrModelName(serviceTarget, Scenario.LegacyCompletions);
+            string promptText = "Are bananas especially radioactive?";
+            var requestOptions = new CompletionsOptions()
+            {
+                DeploymentName = deploymentOrModelName,
+                Prompts = { promptText },
+                Temperature = 0.75f,
+                User = "AzureSDKOpenAITests",
+                Echo = true,
```
Comment (trrwilson) [inline sdk/openai/Azure.AI.OpenAI/tests/CompletionsTests.cs:156]: oh, I should have read a bit further -- it looks like it was just expanded and split. If that's the case, looks like it's all taken care of! (I'll leave active just to confirm, though)

**Review by trrwilson — APPROVED** [review]

File: sdk/openai/Azure.AI.OpenAI/tests/CompletionsTests.cs (hunk 3 above)
Comment (ninghu) [reply sdk/openai/Azure.AI.OpenAI/tests/CompletionsTests.cs:111]: Yeah, setting 'echo' and 'logprobs' at the same time is not supported for `instruct` model, so I seperate it into a new test `AdvancedCompletionsOptionsWithEcho`

File: sdk/openai/Azure.AI.OpenAI/tests/CompletionsTests.cs (hunk 4 above)
Comment (ninghu) [reply sdk/openai/Azure.AI.OpenAI/tests/CompletionsTests.cs:156]: Your understanding is correct. We have a seperated test for covering `echo` case.

File: sdk/openai/Azure.AI.OpenAI/CHANGELOG.md (hunk 1 above)
Comment (ninghu) [reply sdk/openai/Azure.AI.OpenAI/CHANGELOG.md:3]: Make sense. Will remove it.

File: sdk/openai/Azure.AI.OpenAI/README.md (hunk 2 above)
Comment (ninghu) [reply sdk/openai/Azure.AI.OpenAI/README.md:701]: Agreed, but I think we opted for `Input` to stay consistent with OpenAI. 

https://platform.openai.com/docs/guides/text-to-speech?lang=python

```
response = client.audio.speech.create(
  model="tts-1",
  voice="alloy",
  input="Today is a wonderful day to build something people love!"
)
```

**Pull request merged**